airbyte-cdk/python/airbyte_cdk/sources/streams/http/requests_native_auth/declarative_oauth.py

```diff
@@ -0,0 +1,137 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Any, List, Mapping
+
+import pendulum
+from airbyte_cdk.sources.declarative.interpolation.interpolated_mapping import InterpolatedMapping
+from airbyte_cdk.sources.declarative.interpolation.interpolated_string import InterpolatedString
+from airbyte_cdk.sources.streams.http.requests_native_auth.abstract_oauth import AbstractOauth2Authenticator
+
+
+class DeclarativeOauth2Authenticator(AbstractOauth2Authenticator):
```

Comment:
should we move this to the declarative package so the inter-module dependencies only go in one direction?

Reply:
good point. i didn't really consider that, but I think it makes sense to keep interpolation out of existing cdk code as much as possible. will move this